packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java

```diff
@@ -405,6 +410,26 @@ public void updateClippingRect() {
     updateClippingToRect(mClippingRect);
   }
 
+  @Override
+  public void startViewTransition(View view) {
+    super.startViewTransition(view);
+    mIsTransitioning = true;
+  }
+
+  @Override
+  public void endViewTransition(View view) {
+    super.endViewTransition(view);
+    mIsTransitioning = false;
+    mChildrenRemovedWhileTransitioning = null;
```

Comment:
@tdn120 I'm not sure though, what you mean by this:

> Does this account for multiple views/transitions? Seems like we'd need to remove the id(s) instead [...]

Could you elaborate here? 

The `ReactViewGroup` instance is responsible only of tracking its transitioning state & children. I wrote it so to align it with Android `parent.startViewTransition(child)` behaviour - it modifies only the `parent` behaviour with respect to this single `child`. The `child` is unaware of any changes.

Reply:
Just at a basic level, if you have a timeline of:
```
startViewTransition(viewA)
startViewTransition(viewB)
endViewTransition(viewA) 
```
At this point you would clear `mIsTransitioning` and `mChildrenRemovedWhileTransitioning`, so wouldn't `viewB` be out of sync until its transition ends?

Reply:
I was OOO by the end of last week. Back here now 👋🏻 

~I think I still don't understand the issue here.~

~Android docs [explicitly state](https://developer.android.com/reference/android/view/ViewGroup#startViewTransition(android.view.View)) that a call to `startViewTransition` should be accompanied by call to `endViewTransition`. It is left up to the programmer to ensure this.~

~If you - as a API user - fail to fulfil this contract you'll get inconsistent (possibly buggy) behaviour and I believe there is not much we can do on React Native end to prevent that.~

~Regarding the example:~

```java
this.startViewTransition(viewA)
this.startViewTransition(viewB)
this.endViewTransition(viewA) 
```

~...~

Oh, it just got to me. Yeah, I think you're right and that is a very good catch. Thanks. I'll update the code in a bit.

Reply:
I should keep track of which children are transitioning and clear the state only if there are no more such children.

Reply:
@tdn120 I am thinking about various approaches, trying to avoid allocating additional hash set as much as possible. I would need it to store all the children for which `startViewTransition` has been called and then remove them from the set in `endViewTransition`. The working approach would look somehow like that:

```java
  public boolean isChildTransitioning(View child) {
    return mTransitioningChildren != null && mTransitioningChildren.contains(child.getid());
  }

  public void onViewRemoved(View child) {
    ...

    if (isChildTransitioning(child)) {
      // This call would allocate
      ensureChildrenRemovedWhileTransitioning.add(child.getId());
    }

    ...
  }

  public void startViewTransition(View view) {
    super.startViewTransition(view);
    ensureTransitioningChildren().add(view.getId());
  }

  public void endViewTransition(View view) {
    super.endViewTransition(view);
    ensureTransitioningChildren().remove(view.getId());
    if (mChildrenRemovedWhileTransitioning != null) {
      mChildrenRemovedWhileTransitioning.remove(view.getId());
    }
  }
```

However, this requires 2 hash sets...

Only way to improve I see here is to rely in `onViewRemoved` on the fact, that **if children is transitioning and it has been removed then its parent won't be nulled out**. 
Hence we could check it like this:

```java
public void onViewRemoved(View child) {
    ...
    
    if (child.getParent() != null) {
        ensureChildrenRemovedWhileTransitioning.add(child.getId());
    }

    ...
}
```

Looking at Android code [(link)](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/view/ViewGroup.java#5398) this would be reliable.
Even if I'm overlooking some case this still would solve the initial issue, so we should be fine on this front.

Another possibility is to store the information on whether the view is transitioning or not directly on the child view itself, by using a tag as you suggested.

Lemme know which behavior you would agree for.